libcudacxx/include/cuda/std/detail/libcxx/include/__bit_reference

```diff
@@ -137,8 +164,8 @@ inline _LIBCUDACXX_INLINE_VISIBILITY void swap(bool& __x, __bit_reference<_Cp> _
 template <class _Cp>
 class __bit_const_reference
 {
```

Comment:
I would love if we would turn this into a single class
```suggestion
template <class _Cp, bool _IsConst>
class __bit_const_reference
```
And then only SFINAE the few functions that are different. That would be much closer to all the modern ranges iterators and greatly helps localizing the code

Reply:
That would require turning all those members into templates themselves, no? The const version of this is trivial enough that I'd rather not deal with that.